cli/testdataseeder.php

```diff
@@ -132,8 +162,62 @@ function create_test_data($table) {
     create_user_logged_out($table);
 }
 
-// create 10 rows in each table
-create_test_data("logstore_xapi_log");
-create_test_data("logstore_xapi_failed_log");
+/**
+ * Create a user and return the userid.
+ * If the user already exists then return the userid. 
+ *
+ * @param string $username username
+ * @param string $firstname firstname
+ * @param string $lastname lastname
+ * @return int userid
+ */
+function create_user($username, $firstname, $lastname) {
+    global $DB;
+
+    // check if user exists
+    $user = $DB->get_record('user', array('username' => $username), "id,username");
+    if ($user) {
+        return $user->id;
+    }
+
+    // generate user
+    $generator = new testing_data_generator();
+    try {
+        $user = $generator->create_user([
+            'username' => $username,
+            'firstname' => $firstname,
+            'lastname' => $lastname
+        ]);
+    } catch (Exception $e) {
+        echo $e->getMessage().PHP_EOL;
+        return 0;
+    }
+
+    return $user->id;
+}
+
+function create_standing_data() {
+    $user1 = create_user("user1", "User", "One");
+    $user2 = create_user("user2", "User", "Two");
+    echo "UserID 1: ".$user1.PHP_EOL;
+    echo "UserID 2: ".$user2.PHP_EOL;
+
+    // TODO: create course we cannot restore a real course programmatically at the moment
+    // course should contain a quiz, forum and assignment
+    // the forum post and assignment submissions are an added complication
+    // assignment grade is related to the assignment submission
+}
+
+function create_data_set() {
+    // create n-rows in each table
+    // so we get one of each event and pad out the rest with course_viewed events
+    $rows = 10;
```

Comment:
should be defined in the top of the file?

Reply:
Yes that is a better idea.